flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1266,15 +1282,21 @@ public CompletableFuture<Acknowledge> freeSlot(
 
     @Override
     public void freeInactiveSlots(JobID jobId, Time timeout) {
-        log.debug("Freeing inactive slots for job {}.", jobId);
+        MdcUtils.addJobID(jobId);
```

Comment:
Theoretically you could implement this part in a generic way in the InvocationHandler by checking if any of the method arguments is a job ID.

Reply:
It wouldn't work for all cases because sometimes we need to extract job ID from the argument or TM state.
(And also that would be much less readable IMO)

Reply:
yes, it was more of an idea on what we could do in the future.